### ISSUE TITLE
Implement a 0-trust solution to clearweb services

### ISSUE THREAD

**Issue by Kreyren** [Arcanyx-org/NiXium#77]
As discussed in order to provide services for the clearweb we need to implement a solution that makes the servers not geo-locatable as added layer of security to reduce the risk of a physical attack.

### Tor

Tor lacks Post Quantum Cryptography and has too many issues with using centralized relays that makes basically everyone to block it and is too vulnerable to sybil attacks, thus considered unusable and we are currently migrating the intranet on Lokinet in https://github.com/NiXium-org/NiXium/pull/76

### Lokinet

In review..

### SPN by Safing

Good idea, terrible implementation atm. Lacks Post Quantum Safety and seems to be a nightmare for Harvest Now, Decrypt Later.

I did a review in https://github.com/safing/portmaster/issues/1598 and asked upstream to address these issues, if the upstream sucks it might not be a bad idea to fork them.

Also it's currently not implemented in Nixpkgs https://github.com/NixOS/nixpkgs/pull/264454

**DISQUALIFIED:** Rationale in https://github.com/safing/portmaster/issues/1598#issuecomment-2199794781

### VPN

Terrible solution as the VPN service provider can identify the location of the client and can circumvent encryption to monitor the traffic, could be made to work by using a VPS that we connect to over onion/garlic routing, but it's implementation would be complex af and hard to maintain.

## Expected Clearweb Services

* Gitea
* Nextcloud
* Nixpkgs Mirror
* Invidious
* Mumble
* E-Mail
* Jitsi
* F-Droid Repo
* Cydia Repo
* etc..

**Comment by Kreyren** [comment]
@TanvirOnGH attn and review request: SPN disqualified from selection.

**Comment by ghost** [comment]
I agree that SPN by Safing is currently not a viable option due to its lack of PQS and the other issues outlined. The disqualification seems justified based on the rationale provided.

Given the challenges with other options like Tor and VPNs, it seems that Lokinet might be our best bet if it passes the review. We should continue to explore Lokinet as a solution for providing clearweb services securely.

I'll look into the progress on Lokinet and see if there are any updates or additional considerations we should be aware of. If anyone has alternative suggestions or further insights, please feel free to share.
